src/components/SimpleSlider.js

```diff
@@ -0,0 +1,176 @@
+const React = require('react');
+const ReactDOM = require('react-dom');
+const Radium = require('radium');
+const _merge = require('lodash/merge');
+
+const StyleConstants = require('../constants/Style');
+
+const SimpleSlider = React.createClass({
+  propTypes: {
+    disabled: React.PropTypes.bool,
+    onPercentChange: React.PropTypes.func.isRequired,
+    percent: React.PropTypes.number.isRequired,
+    selectedColor: React.PropTypes.string,
+    styles: React.PropTypes.object
+  },
+
+  getDefaultProps () {
+    return {
+      selectedColor: StyleConstants.Colors.PRIMARY
+    };
+  },
+
+  getInitialState () {
+    return {
+      dragging: false,
+      leftPixels: 0,
+      width: 0
+    };
+  },
+
+  componentDidMount () {
+    this._setDefaults();
+  },
+
+  componentWillReceiveProps (newProps) {
+    const leftPixels = (newProps.percent * this.state.width);
+
+    this.setState({ leftPixels });
+  },
+
+  _setDefaults () {
+    const component = ReactDOM.findDOMNode(this.rangeSelectorRef);
+    const componentStyles = window.getComputedStyle(component);
+    const width = parseInt(componentStyles.width, 0);
+    const leftPixels = this.props.percent * width;
+
+    this.setState({ width, leftPixels });
+  },
+
+  _handleMouseEvents (e) {
+    const clientX = e.touches ? e.touches[0].clientX : e.clientX;
+    const leftSpace = ReactDOM.findDOMNode(this.rangeSelectorRef).getBoundingClientRect().left;
+    let currentPercent = ((clientX - leftSpace) / this.state.width);
+
+    if (currentPercent < 0) {
+      currentPercent = 0;
+    } else if (currentPercent > 1) {
+      currentPercent = 1;
+    }
+
+    this.props.onPercentChange(currentPercent);
+  },
+
+  _handleTrackMouseDown (e) {
+    this._handleMouseEvents(e);
+  },
+
+  _handleDragStart () {
+    this.setState({
+      dragging: true
+    });
+  },
+
+  _handleDragging (e) {
+    if (this.state.dragging) {
+      this._handleMouseEvents(e);
+    }
+  },
+
+  _handleDragEnd () {
+    this.setState({
+      dragging: false
+    });
+  },
+
+  render () {
+    const styles = this.styles();
+
+    return (
+      <div style={styles.component}>
+        <div
+          onMouseLeave={this.props.disabled ? null : this._handleDragEnd}
```

Comment:
instead of typing `this.props.disabled` over and over again, can we move 
`const { disabled } = this.props;` to the top and then use `disabled` below?